WM_TAKE_FOCUS support

gloggers99 opened the pull request:
Explicitly send a WM_TAKE_FOCUS message (if supported) to the newly focused window. This should fix most input focus based issues and maybe even the macopix focus related issue in the readme. I have tested this solution on JetBrains based IDE's and it works.

For the restart function passing NULL instead of a parameter list is undefined behavior.

venam commented:
Hello @gloggers99 ,
That's a great idea, thanks for helping out! Hopefully this will fix a couple of issues you've mentioned.

I got just a small detail that could improve this, let's add in the `struct client` in `types.h` an integer or enum that would indicate whether this client needs WM_TAKE_FOCUS and if we've already asked if it supports it. Like that we won't have to constantly re-query the same client each time and will only have to check this value. This would save some back and forth.  
This could also be added on restart or mapping, but that might be more of a pain to do.

If you could also squash the commits afterward that would be fantastic!
Thanks!

gloggers99 commented:
Hello @venam, I went ahead and added the field to the client struct and added the boolean check in the focusing function! Let me know what you think.

I believed I squashed the commits properly, I wasn't familiar with commit squashing.

venam commented:
Hello,
It looks solid now with the rectifications.

> I believed I squashed the commits properly, I wasn't familiar with commit squashing.

There are two commits, the idea is to squash them both into one so that it's cleaner when merging. Can you please do that and I'll merge.

Thanks a lot!
